pkg/cli/admin/release/mirror_test.go

```diff
@@ -4,116 +4,242 @@ import (
+					mirrors: []string{"registry/ocp/test"},
+				},
+				{
+					source:  "quay/ocp/test",
+					mirrors: []string{"registry/another/test"},
+				},
+				{
+					source:  "quay/ocp/test",
+					mirrors: []string{"registry/ocp/test"},
+				},
+			},
+			expectedSources: []apicfgv1.ImageDigestMirrors{
 				{
 					Source:  "quay/ocp/test",
-					Mirrors: []string{"registry/ocp/test"},
+					Mirrors: []apicfgv1.ImageMirror{"registry/ocp/test"},
 				},
 				{
 					Source:  "quay/ocp/test",
-					Mirrors: []string{"registry/another/test"},
+					Mirrors: []apicfgv1.ImageMirror{"registry/another/test"},
 				},
 				{
-					Source:  "quay/another/test",
-					Mirrors: []string{"registry/ocp/test"},
+					Source:  "quay/ocp/test",
+					Mirrors: []apicfgv1.ImageMirror{"registry/ocp/test"},
 				},
+			},
+		},
+		{
+			name: "multiple Source single Mirrors",
+			sources: []mirrorSet{
 				{
-					Source:  "quay/another/test",
-					Mirrors: []string{"registry/another/test"},
+					source:  "quay/ocp/test",
+					mirrors: []string{"registry/ocp/test"},
+				},
+				{
+					source:  "quay/another/test",
+					mirrors: []string{"registry/ocp/test"},
 				},
 			},
-			expectedSources: []operatorv1alpha1.RepositoryDigestMirrors{
+			expectedSources: []apicfgv1.ImageDigestMirrors{
+				{
+					Source:  "quay/ocp/test",
+					Mirrors: []apicfgv1.ImageMirror{"registry/ocp/test"},
+				},
 				{
 					Source:  "quay/another/test",
-					Mirrors: []string{"registry/ocp/test", "registry/another/test"},
+					Mirrors: []apicfgv1.ImageMirror{"registry/ocp/test"},
+				},
+			},
+		},
+		{
+			name: "multiple Source multiple Mirrors",
+			sources: []mirrorSet{
+				{
+					source:  "quay/ocp/test",
+					mirrors: []string{"registry/ocp/test"},
+				},
+				{
+					source:  "quay/ocp/test",
+					mirrors: []string{"registry/another/test"},
+				},
+				{
+					source:  "quay/another/test",
+					mirrors: []string{"registry/ocp/test"},
+				},
+				{
+					source:  "quay/another/test",
+					mirrors: []string{"registry/another/test"},
+				},
+			},
+			expectedSources: []apicfgv1.ImageDigestMirrors{
+				{
+					Source:  "quay/ocp/test",
+					Mirrors: []apicfgv1.ImageMirror{"registry/ocp/test"},
 				},
 				{
 					Source:  "quay/ocp/test",
-					Mirrors: []string{"registry/ocp/test", "registry/another/test"},
+					Mirrors: []apicfgv1.ImageMirror{"registry/another/test"},
+				},
+				{
+					Source:  "quay/another/test",
+					Mirrors: []apicfgv1.ImageMirror{"registry/ocp/test"},
+				},
+				{
+					Source:  "quay/another/test",
+					Mirrors: []apicfgv1.ImageMirror{"registry/another/test"},
 				},
 			},
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			uniqueSources := dedupeSortSources(tt.sources)
-			if !reflect.DeepEqual(uniqueSources, tt.expectedSources) {
-				t.Errorf("%s", diff.ObjectReflectDiff(uniqueSources, tt.expectedSources))
+			idmsSources := convertMirrorSetToImageDigestMirrors(tt.sources)
```

Comment:
Where is the old icsp tests?

Reply:
since there's no separate convert function needed for ICSP, I added tests for printICSPInstructions and printICSPInstructions